blog/_posts/2021-07-08-portforwarding-is-awesome.md

```diff
@@ -0,0 +1,292 @@
+---
+slug: "portforwarding-is-awesome"
+date: 2021-07-08
+author: nick
+layout: blog
+title: "Portforwarding should be a tool in every dev toolbox"
+subtitle: "An overview of socat, kubectl port-forward, and how Tilt manages portforwards"
+description: "An overview of socat, kubectl port-forward, and how Tilt manages portforwards"
+image: "/assets/images/portforwarding-is-awesome/tunnel.jpg"
+image_caption: "The Joralemon Street Tunnel, via <a href='https://commons.wikimedia.org/wiki/File:Joralemon_Street_Tunnel_postcard,_1913.jpg'>commons.wikipedia.org</a>."
+tags:
+  - api
+  - networking
+---
+
+In the Unix mindset, every program reads and writes files. So it's important to have a toolkit of basic
+tools like `cat` for working with files!
+
+In 2021, we mostly write servers. That's why it's important to have a toolkit for working with 
+network sockets!
+
+In this post, we're going to do a quick tour of the tools we (Tilt-team) work with everyday,
+how we use them, and how we're trying to make it easier to manage them.
+
+## Reverse-engineering multi-service apps with `socat`
+
+I really like Cindy Sridharan's introduction to `socat`:
+
+> `socat` truly is the Swiss Army Knife of network debugging tools.
+
+> `socat` stands for SOcket CAT. It is a utility for data transfer between two addresses.
+
+> I personally use `socat` not so much as a production debugging tool than for
+> troubleshooting local development issues (especially when developing network
+> services).
+
+Go ahead and [read it](https://copyconstruct.medium.com/socat-29453e9fc8a6) and come back. I'll wait.
+
+.
```

Comment:
Realistically, no one's gonna go read it right away. People will open a new tab that'll sit there for later. We can omit this multiline "ellipsis."

Reply:
done